code/__DEFINES/preferences.dm

```diff
@@ -120,3 +147,34 @@
 #define HELMET_DEFAULT "Default"
 #define HELMET_MK2 "Mark II"
 #define HELMET_PROTECTIVE "Protective"
+
+// All DB preference entries go here
+// --- DO NOT EVER CHANGE OR RE-USE VALUES HERE ---
+// If you remove an entry, comment it out and leave it for preservation sake
+// All the values must be strings because they are map entries not list indexes
+#define PREFERENCE_TAG_TOGGLES			"1"
```

Comment:
If these all have to be strings, why not use actual names instead of numbers?

Reply:
It gets stored in the DB as an int itself which results in less space used than storing `1` as a string. Its just a string in BYOND because BYOND cant do integer map-indexes, it only uses it for nth-place indexes.

Reply:
Right, but I guess my question (implicitly) was, why not change the DB schema to VARCHAR(20)? Ints are 4 bytes each. A VARCHAR(20) is (at most) 21 bytes. 1000 users * 100 different prefs * 21 bytes = 2.1MB, which is absolutely nothing. You don't need to be scrimping for space here.

In contrast, if you have descriptive names for the pref keys, it will be much easier to look stuff up and generally operate on the data in the DB.

Reply:
> Right, but I guess my question (implicitly) was, why not change the DB schema to VARCHAR(20)? Ints are 4 bytes each. A VARCHAR(20) is (at most) 21 bytes. 1000 users * 100 different prefs * 21 bytes = 2.1MB, which is absolutely nothing. You don't need to be scrimping for space here.
> 
> In contrast, if you have descriptive names for the pref keys, it will be much easier to look stuff up (manually) and generally operate on the data in the DB.

@Crossedfall want your input here.

Reply:
One other thing I'll mention while we wait for Crossedfall: If this is an amount of space to be concerned about, you'll probably have to denorm the schema (again), because the ckeys are much bigger than this and are also being duplicated for each pref.

Reply:
Make it bitflags

Reply:
Why the hell would we make this bitflags? Theyre not entries to be added together, theyre unique things.